### DISCUSSION TITLE
Makefile linking options ( undefined reference to `dlopen' error )

### DISCUSSION THREAD

**Discussion by mrfloydst** [shorepine/alles#61] [General]
Hi, I found that Alles (software version) doesn't compile on a Raspberry PI atm unless you add -ldl to the LIBS line in the Makefile. If you don't do that, the compiler will throw a "undefined reference to `dlopen'" error.
Hope this helps someone.

**Comment by bwhitman** [comment]
Yes, we fixed this in AMY with 

```
# on Raspberry Pi, at least under 32-bit mode, libatomic and libdl are needed.
ifeq ($(shell uname -m), armv7l)
LIBS += -ldl  -latomic
endif
```

In the Makefile. we're working on making AMY a submodule of Alles, which will help these stay in sync. See #60 for that progress. But for now, you can update this Makefile no problem!

**Comment by bwhitman** [comment]
I've made this change to the Alles makefile. I don't have a Rpi 32-bit to test on, but give it a go if you can!

**Reply by mrfloydst** [reply]
Hello Brian,
it compiles on 32bit PI now, but not on aarch64 - for that, I'd add
ifeq ($(shell uname -m), aarch64)
        LIBS += -ldl  -latomic
endif

Off topic - did you get my mail? I'd love to do a video on this project, and if you happen to have some more photos or videos on the actual product and are willing to share them for that use, that'd be great. ;-)
